processor/awscontainerinsightprocessor/processor.go

```diff
@@ -0,0 +1,121 @@
+// Copyright  OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package awscontainerinsightprocessor
+
+import (
+	"context"
+	"errors"
+	"os"
+	"time"
+
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/awscontainerinsightprocessor/internal/stores"
+)
+
+type awscontainerinsightprocessor struct {
+	logger                  *zap.Logger
+	stores                  []stores.K8sStore
+	storeRefreshMinInterval time.Duration
+	storeRefreshTimeout     time.Duration
+	storeRefreshStoppedC    chan bool //used only by test
+	shutdownC               chan bool
+	TagService              bool   `toml:"tag_service"`
+	ClusterName             string `toml:"cluster_name"`
+	HostIP                  string `toml:"host_ip"`
+	NodeName                string `toml:"node_name"`
+	PrefFullPodName         bool   `toml:"prefer_full_pod_name"`
+}
+
+// ProcessMetrics takes pdata.Metrics and does metric decorations (e.g. adding resource attributes/new metrics)
+func (acip *awscontainerinsightprocessor) ProcessMetrics(_ context.Context, md pdata.Metrics) (pdata.Metrics, error) {
+	//TODO: add logic to do the metric decoration by adding new resource attributes and datapoints
+	return md, nil
+}
+
+// Shutdown stops the aws container insights processor
+func (acip *awscontainerinsightprocessor) Shutdown() {
+	close(acip.shutdownC)
+}
+
+func (acip *awscontainerinsightprocessor) start() error {
+	acip.NodeName = os.Getenv("HOST_NAME")
+	if acip.NodeName == "" {
+		return errors.New("missing environment variable HOST_NAME. Please check your deployment YAML file")
+	}
+
+	acip.HostIP = os.Getenv("HOST_IP")
+	if acip.HostIP == "" {
+		return errors.New("missing environment variable HOST_IP. Please check your deployment YAML file")
+	}
+
+	acip.shutdownC = make(chan bool)
+
+	// TODO: add the relevant store to acip.stores
+	ctx, cancel := context.WithTimeout(context.Background(), acip.storeRefreshTimeout)
+	go func(ctx context.Context, cancel func()) {
+		for _, store := range acip.stores {
+			store.RefreshTick(ctx)
+		}
+		cancel()
+	}(ctx, cancel)
+	<-ctx.Done()
+	cancel()
```

Comment:
Rather than duplicate this code here and inside the wait loop below, can we extract this into a method that can be invoked?